hnn_core/cell_response.py

```diff
@@ -331,6 +332,8 @@ def plot_spikes_raster(self, trial_idx=None, ax=None, show=True):
             Index of trials to be plotted. If None, all trials plotted.
         ax : instance of matplotlib axis | None
             An axis object from matplotlib. If None, a new figure is created.
+        start_from_t0 : bool
```

Comment:
I suppose I'm not sure I understand when this shouldn't be true, do the plots look weird or get cut off?

Reply:
Suppose the first spike occurs at t=50, matplotlib will probably set the xlim to somewhere close but less than 50 like 40. But one may want to plot the whole range from the very beginning of the simulation even if it's not aligned with other subplots, which I guess is the original motivation of `ax.set_xlim(left=0)`. And if we always turn on this option, the alignment will be influenced and all other subplots will start from t=0. Since I personally tend to leave some space for both sides for all plots so I added this behavior.  It indeed looks a bit weird and we can safely remove it if we adopt Mainak's proposal.